### ISSUE TITLE
Convergence Inquiry

### ISSUE THREAD

**Issue by Larkinnjm1** [savinienb/Matwo-CapsNet#2]
Hi folks,

Firsts thanks for the great work on this much appreciated. 

Just a quick question I am working at the minute to implement your multiclass SegCaps model. Did you get it to converge for any of your models? and if so did you utilise a learning rate of 1 to achieve this?

Also I have noted two /three items of interest specifically being the:
-learning rate is hard coded at 1 in main_train_and_test.py line 106
-update_routing,_body method can fail to be loaded appropriately due to line 358 "route: [batch, input_dim, output_dim, ...]" not being able to be read in by the gast library 
-The validation loss function does not take into account what type of loss function is being utilised hence the arguments can be off if you want to use different class weights to the ones specified.

I have fixed these in my own copy of this repo and could do a PR if you wish?

Kind regards,

Niall

**Comment by savinienb** [comment]
Hi,

Despite our best efforts, we couldn't make SegCaps converge in an appropriate time frame. Based on the slow convergence of binary SegCaps when trained on binary cases, as well as a Multi-Labels SegCaps network which was still learning after 1 000 000 training iterations, it might be possible than training the networks for many more iterations might lead to satisfactory results.

For the first items of interest:
We purposefully used the Adadelta optimizer to avoid tuning the learning rate (hence the hard coded 1).

I would enjoy seeing your PR as well as the improvement you made to the code.

Last but not least, thanks for your interest in the paper as well as your care for the implementation :) 

Cheers,
Savinien
